Source/monster.cpp

```diff
@@ -4758,69 +4758,43 @@ bool LineClearMissile(Point startPoint, Point endPoint)
 
 bool LineClear(bool (*Clear)(int, Point), int entity, Point startPoint, Point endPoint)
 {
-	int d;
-	int xincD, yincD, dincD, dincH;
-	bool done = false;
-
 	Point position = startPoint;
 
-	int dx = endPoint.x - position.x;
-	int dy = endPoint.y - position.y;
-	if (abs(dx) > abs(dy)) {
-		if (dx < 0) {
-			std::swap(position, endPoint);
-			dx = -dx;
-			dy = -dy;
-		}
-		if (dy > 0) {
-			d = 2 * dy - dx;
-			dincD = 2 * dy;
-			dincH = 2 * (dy - dx);
-			yincD = 1;
+	int d1 = endPoint.x - position.x;
+	int d2 = endPoint.y - position.y;
+
+	bool horizontal = abs(d1) > abs(d2);
+
+	if (!horizontal)
+		std::swap(d1, d2);
+
+	if (d1 < 0) {
+		std::swap(position, endPoint);
+		d1 = -d1;
+		d2 = -d2;
+	}
+
+	int incD = (d2 > 0) ? 1 : -1;
+	int d = 2 * d2 - (d1 * incD);
+	int dincD = 2 * d2;
+	int dincH = 2 * (d2 - (d1 * incD));
+
+	while (position != endPoint && (position == startPoint || Clear(entity, position))) {

```

Comment:
What is the expectation of `Clear`? Shouldn't we rename it to `IsClear`?

Reply:
It checks the individual points in the line. If we wanted to follow the same naming as the function then it should be called `PointClear`, else we should also rename LineClear to `IsLineClear` and the same for all the related functions that follow the same naming. But I think such a change belongs in it's own PR so we don't clutter this with multiple files and case conflict hell for this very isolated PR.